.golangci.yaml

```diff
@@ -1,8 +1,42 @@
+# Golang CI pipeline configuration
+linters:
+  disable-all: true
+
+  # Run golangci-lint linters to see the list of all linters
+  # Please keep them sorted alphabetically
+  enable:
+    - bodyclose
+    - deadcode
+    - depguard
+    - dogsled
+    - errcheck
+    - goconst
+    - gocritic
+    - gofmt
+    - goimports
+    - goprintffuncname
+#    - gosec
+    - gosimple
+    - govet
+    - ineffassign
+    - misspell
+    - nakedret
+    - nolintlint
+    - revive
+    - rowserrcheck
+    - staticcheck
+    - structcheck
+    - stylecheck
+    - typecheck
+    - unconvert
+#    - unparam
+    - varcheck
+    - whitespace
+
 linters-settings:
   gocritic:
     disabled-checks:
       # TODO: re-enable after figuring out why runcommand fails on go-critic..
       - unlambda
-linters:
-  enable:
-    - gocritic
\ No newline at end of file
+  goconst:
+    min-len: 5
```

Comment:
I'm a bit wary of custom settings for the linters* - what's the rationale for changing it? Do we have a lot of strings that would otherwise warn and if so, why not constify them if we want goconst enabled?

*) Two reasons - copypasting custom settings across different codebases will make it easy to leave them unmaintained and consistency across our and all other codebases (although, with goconst, I guess it's not that important)

Reply:
though that we can keep the golangci-lnit config file consistent across our projects. I just copy paste it from the terraform provider.